tests/pytests/integration/ssh/test_ssh_setup.py

```diff
@@ -0,0 +1,246 @@
+        base_script_args=["--ignore-host-keys"],
+    )
+
+
+def test_setup(salt_ssh_cli, ssh_container_name, ssh_sub_container_name, ssh_password):
+    """
+    Test salt-ssh grains id work for localhost.
+    """
+    # Provide the passwd from the CLI to allow the key deploy
+    possible_ids = (ssh_container_name, ssh_sub_container_name)
+    ret = salt_ssh_cli.run(
+        "--passwd", ssh_password, "--key-deploy", "grains.get", "id", minion_tgt="*"
+    )
+    assert ret.returncode == 0
+    for id in possible_ids:
+        assert id in ret.data
+        assert ret.data[id] == id
+
+    # Run it again without the key deploy
+    ret = salt_ssh_cli.run("grains.get", "id", minion_tgt="*")
+    assert ret.returncode == 0
+    for id in possible_ids:
+        assert id in ret.data
+        assert ret.data[id] == id
+
+    # Run a test.sleep and kill it
+    sleep_time = 15
+    cmdline = salt_ssh_cli.cmdline("test.sleep", sleep_time, minion_tgt="*")
+    terminal_stdout = tempfile.SpooledTemporaryFile(512000, buffering=0)
+    terminal_stderr = tempfile.SpooledTemporaryFile(512000, buffering=0)
+
+    proc = subprocess.Popen(
+        cmdline,
+        shell=False,
+        stdout=terminal_stdout,
+        stderr=terminal_stderr,
+        universal_newlines=True,
+    )
+    start = time.time()
+    try:
+        # Make sure it actually starts
+        proc.wait(1)
+    except subprocess.TimeoutExpired:
+        pass
+    else:
+        terminate_process(proc.pid, kill_children=True)
+        pytest.fail("The test process failed to start")
+
+    time.sleep(2)
+    # Send CTRL-C to the process
+    os.kill(proc.pid, signal.SIGINT)
+    with proc:
+        # Wait for the process to terminate, to avoid zombies.
+        # Shouldn't really take the 30 seconds
+        proc.wait(sleep_time * 2)
+        # poll the terminal so the right returncode is set on the popen object
+        proc.poll()
+        # This call shouldn't really be necessary
+        proc.communicate()
+    stop = time.time()
+
+    terminal_stdout.flush()
+    terminal_stdout.seek(0)
+    if sys.version_info < (3, 6):  # pragma: no cover
+        stdout = proc._translate_newlines(
+            terminal_stdout.read(), __salt_system_encoding__
+        )
+    else:
+        stdout = proc._translate_newlines(
+            terminal_stdout.read(), __salt_system_encoding__, sys.stdout.errors
+        )
+    terminal_stdout.close()
+
+    terminal_stderr.flush()
+    terminal_stderr.seek(0)
+    if sys.version_info < (3, 6):  # pragma: no cover
+        stderr = proc._translate_newlines(
+            terminal_stderr.read(), __salt_system_encoding__
+        )
+    else:
+        stderr = proc._translate_newlines(
+            terminal_stderr.read(), __salt_system_encoding__, sys.stderr.errors
+        )
+    terminal_stderr.close()
+    ret = ProcessResult(
+        returncode=proc.returncode, stdout=stdout, stderr=stderr, cmdline=proc.args
+    )
+    log.debug(ret)
```

Comment:
is there any possibility this would print out a password?

Reply:
This one won't, but I know the first salt-ssh command
```
ret = salt_ssh_cli.run(
    "--passwd", ssh_password, "--key-deploy", "grains.get", "id", minion_tgt="*"
)
```
will log the password.  However, this is a one-time use password that's randomly generated and is invalidated as soon as the containers (that are only used for this single test) are torn down.

Reply:
Someone could then access that container while its running. Can we ensure we aren't logging the password.